src/qibo/measurements.py

```diff
@@ -117,12 +124,12 @@ def add_shot_from_sample(self, sample):
     def has_samples(self):
         return self._samples is not None
 
-    def register_samples(self, samples, backend=None):
+    def register_samples(self, samples):
         """Register samples array to the ``MeasurementResult`` object."""
-        self._samples = samples
+        self._samples = self.backend.cast(samples, self.backend.np.int64)
```

Comment:
I would avoid casting if not strictly needed...

#1412 #1414 #1417

Reply:
Yeah, I agree, the problem though is whether the samples passed to the `register_samples` method are not proper tensors of the `MeasurementResult.backend`. At that point, you either raise an error or cast.

Reply:
> Yeah, I agree, the problem though is whether the samples passed to the `register_samples` method are not proper tensors of the `MeasurementResult.backend`. At that point, you either raise an error or cast.

It could be a good reason in favor of dropping `MeasurementResult.backend`...